File: LeetCode/1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        int n = nums.length;

        Arrays.sort(nums);

        int minimizedMaxPairSum = 0;

        for(int i = 0; i < n/2; i++) {
            int sum = nums[i] + nums[n - i - 1];
            minimizedMaxPairSum = Math.max(minimizedMaxPairSum, sum);
        }

        return minimizedMaxPairSum;
    }
}

File: LeetCode/1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.go
func minPairSum(nums []int) int {
    n := len(nums)

    slices.Sort(nums)

    minimizedMaximumPairSum := 0;

    for i := 0; i < n/2; i++ {
        sum := nums[i] + nums[n - i - 1]
        if minimizedMaximumPairSum < sum {
            minimizedMaximumPairSum = sum
        }
    }

    return minimizedMaximumPairSum
}